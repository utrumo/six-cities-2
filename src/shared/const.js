const MAX_IMAGE_ON_OFFER_PAGE = 6;
const ASSETS_PATCH = `/`;

const OfferTypeToPresentName = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export {
  MAX_IMAGE_ON_OFFER_PAGE,
  ASSETS_PATCH,
  OfferTypeToPresentName
};
