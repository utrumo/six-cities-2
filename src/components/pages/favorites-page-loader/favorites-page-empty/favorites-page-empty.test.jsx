import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {MemoryRouter as Router} from 'react-router-dom';
import NameSpace from 'store/name-spaces.js';
import {DEFAULT_NUMBER_VALUE, SortingVariants} from 'shared/const.js';
import FavoritesPageEmpty from './favorites-page-empty.jsx';

it(`FavoritesPageEmpty should renders correctly`, () => {
  const mockStore = configureStore();
  const store = mockStore({
    [NameSpace.DATA]: {
      offers: [],
      offersReviews: [],
      currentLocation: ``,
      currentOfferId: DEFAULT_NUMBER_VALUE,
      sortOrder: SortingVariants.POPULAR,
    },
    [NameSpace.USER]: {
      isAuthorized: false,
      emailValidationError: ``,
      profile: {},
    },
  });
  const favoriteLocationsItems = renderer.create(
      <Provider store={store}>
        <Router>
          <FavoritesPageEmpty />
        </Router>
      </Provider>
  ).toJSON();
  expect(favoriteLocationsItems).toMatchSnapshot();

});
