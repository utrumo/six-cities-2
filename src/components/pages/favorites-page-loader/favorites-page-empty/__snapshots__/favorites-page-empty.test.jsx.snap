// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPageEmpty should renders correctly 1`] = `
<div
  className="page page--favorites-empty"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            aria-current="page"
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "cursor": "default",
              }
            }
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
